dotCMS/src/main/java/com/dotmarketing/portlets/hostvariable/ajax/HostVariableAjax.java

```diff
@@ -1,129 +1,192 @@
 package com.dotmarketing.portlets.hostvariable.ajax;
 
-import java.util.Date;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-
-import javax.servlet.http.HttpServletRequest;
-
 import com.dotcms.repackage.org.directwebremoting.WebContext;
 import com.dotcms.repackage.org.directwebremoting.WebContextFactory;
-
 import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.NoSuchUserException;
 import com.dotmarketing.business.web.UserWebAPI;
 import com.dotmarketing.business.web.WebAPILocator;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotRuntimeException;
 import com.dotmarketing.exception.DotSecurityException;
-import com.dotmarketing.portlets.contentlet.business.HostAPI;
 import com.dotmarketing.portlets.hostvariable.bussiness.HostVariableAPI;
 import com.dotmarketing.portlets.hostvariable.model.HostVariable;
+import com.dotmarketing.util.Logger;
 import com.dotmarketing.util.RegEX;
 import com.dotmarketing.util.UtilMethods;
 import com.liferay.portal.PortalException;
 import com.liferay.portal.SystemException;
 import com.liferay.portal.language.LanguageUtil;
 import com.liferay.portal.model.User;
 
-public class HostVariableAjax {
+import javax.servlet.http.HttpServletRequest;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
 
-	protected HostAPI hostAPI = APILocator.getHostAPI();
+/**
+ * This class allows the interaction with Site Variables in dotCMS. The methods exposed here are referenced via DWR by
+ * the UI layer to allow users to perform CRUD operations on them.
```

Comment:
why not use a Rest end point?

Reply:
This is a legacy class, the core of its code must remain. Otherwise, a lot of code changes are required.

Reply:
I do understand it is a DWR and it is ok, however some of us (R&D) should be in charge about it a migrate the service to an endpoint as Freddy said, we should get far away from DWR as possible